packages/doppler-v3-indexer/src/api/handlers/metadata.ts

```diff
@@ -0,0 +1,95 @@
+import PinataSDK from '@pinata/sdk';
+import { z } from 'zod';
+import type { Context } from 'hono';
+
+// Initialize Pinata
+const pinata = new PinataSDK(
+  process.env.PINATA_API_KEY!,
+  process.env.PINATA_API_SECRET!
+);
+
+// Enhanced validation schema matching PRD specs
+const metadataSchema = z.object({
+  name: z.string().min(1).max(100),
+  symbol: z.string().min(1).max(20),
+  description: z.string().max(1000),
+  image: z.string(), // URL or IPFS hash
+  walletAddress: z.string().regex(/^0x[a-fA-F0-9]{40}$/),
+  socials: z.object({
+    twitter: z.string().optional(),
+    telegram: z.string().optional(),
+    website: z.string().optional(),
+    discord: z.string().optional(),
```

Comment:
i think we don't need this one, it's not in the interface: https://holograph-launchpad.vercel.app/create